### PR TITLE
Upgrade Spezi to 0.8.0

### PR DESCRIPTION
**Pull request by Supereg**
# Upgrade Spezi to 0.8.0

## :recycle: Current situation & Problem
This PR upgrades to Spezi 0.8.0 and moves to observable.

As discussed, this PR removes the Questionnaire Standard Constraint, as we identified that questionnaire responses typically require local context (e.g., the current selected patient). Therefore, questionnaire completion shall always be handled through the closure.

## :gear: Release Notes 
* Migrate to Observable
* Upgrade to Spezi 0.8.0
* Introduce a new `QuestionnaireResult` type, received via the closure
* Removed the Standard constraints.

## :books: Documentation
Documentation was updated and respective parts were removed.

## :white_check_mark: Testing
--

## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: Sources/SpeziQuestionnaire/QuestionnaireView.swift [hunk 1]
```diff
@@ -51,7 +51,7 @@ public struct QuestionnaireView: View {
                 isPresented: $isPresented,
                 questionnaireResponse: { response in
                     await questionnaireResponse?(response)
-                    await questionnaireDataSource.add(response)
+                    await questionnaireDataSource.add(response) // TODO: remove that thingy
```
Comment (PSchmiedmayer) [inline Sources/SpeziQuestionnaire/QuestionnaireView.swift:54]: There is still a TODO in here. Happy to discuss what we should do here in a few minutes.

File: Sources/SpeziQuestionnaire/QuestionnaireView.swift [hunk 2]
```diff
@@ -75,7 +75,7 @@ public struct QuestionnaireView: View {
         questionnaireResponse: (@MainActor (QuestionnaireResponse) async -> Void)? = nil
     ) {
         self.questionnaire = questionnaire
-        self._isPresented = isPresented
+        self._isPresented = isPresented // TODO: this is not optional!
```
Comment (PSchmiedmayer) [inline Sources/SpeziQuestionnaire/QuestionnaireView.swift:78]: There is still a TODO in here. Happy to discuss what we should do here in a few minutes.

**Review by PSchmiedmayer — APPROVED** [review]
Thank you, feel free to merge once we discussed the TODOs and addressed them.

File: Sources/SpeziQuestionnaire/QuestionnaireResult.swift [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+// SPDX-FileCopyrightText: 2023 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import ModelsR4
+
+
+/// The result of a questionnaire.
+public enum QuestionnaireResult {
```
Comment (PSchmiedmayer) [inline Sources/SpeziQuestionnaire/QuestionnaireResult.swift:13]: Nice, looks good!

**Review by PSchmiedmayer — APPROVED** [review]
Thank you for the improvements @Supereg; looks good to me and happy to see it merged 🚀

**Comment by Supereg** [comment]
We have a slightly reduced diff coverage due to the new state handling which is not fully covered by the UI tests (see GitHub review diff). Would it make sense to force merge @PSchmiedmayer?

**Comment by PSchmiedmayer** [comment]
Sure, merged @Supereg 👍

**Pull request merged**